### PR TITLE
[SPARK-35639][SQL][FOLLOWUP] Make hasCoalescedPartition return true if something was actually coalesced

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

Add `CoalescedPartitionSpec(0, 0, _)` check if a `CoalescedPartitionSpec` is coalesced.

### Why are the changes needed?

Fix corner case.

### Does this PR introduce _any_ user-facing change?

yes, UI may be changed

### How was this patch tested?

Add test

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala [hunk 1]
```diff
@@ -92,6 +92,8 @@ case class CustomShuffleReaderExec private(
    */
   def hasCoalescedPartition: Boolean = {
     partitionSpecs.exists {
+      // shuffle from empty RDD
+      case CoalescedPartitionSpec(0, 0, _) => true
```
Comment (ulysses-you) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:96]: it should be more clearer than `s.startReducerIndex + 1 != s.endReducerIndex`

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master/3.2!
